src/components/EntityPage/Participant/Utils/Holder.js

```diff
@@ -0,0 +1,103 @@
+import React from "react";
+import {Container} from "../../../FileRepoSidebar/ui";
+import { Link } from 'react-router-dom';
+import styled from "react-emotion";
+import {withTheme} from "emotion-theming";
+
+//taken from secondaryNavMenu
+const MenuWrapper = styled('ul')`
+  ${({ theme }) => theme.secondaryNav}
+`;
+
+class Holder extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    //wrap the item in a one-element array if the item is not itself an array
+    if(!(props.children instanceof Array)) {
+      this.tabIDs = [props.children.props.label];
+      this.children = [props.children];
+    } else {
+      this.tabIDs = props.children.map( child => child.props.label );
+      this.children = props.children;
+    }
+
+    this.state = {activeTab: this.tabIDs[0]};
+  }
+
+  /**
+   * Makes the buttons for the tabs.
+   *
+   * @param holder This, as it is needed inside a nested class
+   * @returns {*|Array|*[]}
+   */
+  makeTabButtons(holder) {
+    return this.tabIDs.map(tabId => {
+
+      const isActive = tabId === this.state.activeTab;
+
+      //needs to be a class to use setState to do the hovering. Would do in CSS, but with emotion it's too complicated and hacky
+      class Button extends React.Component {
+        constructor(props) {
+          super(props);
+
+          this.state = {hovered: false};
+        }
+
+        render() {
+          //uses a fake Link: we want the same style as a SecondaryNavMenu, and this is the key to it
+          return (
+            <li
+              style={
+                {
+                  borderRight: (isActive || this.state.hovered) ? "5px solid #e83a9c" : "5px solid transparent"
+                }
+              }
+
+              onClick={ () => holder.setState({activeTab: tabId})}
```

Comment:
You should use `this`. To have `this` being bound to the correct value, use `bind`.

i.e. `this.makeTabButtons = this.makeTabButtons.bind(this)` in the constructor.

If you end up doing that more than once in the same class, use the `autobind` instead:
```
import autobind from 'auto-bind-es5';
...
constructor(props) {
  super(props);
   ...
  autobind(this);
}
```

It basically does the above on every method of an object. It's used in a few places in this project.

Reply:
You should not have the need to. Affecting the state of a parent directly is a big no-no.

See: https://reactpatterns.com/#state-hoisting

Reply:
Also forgot to mentioned: you should add a `key` property your Button, which should:

- be unique to that button within it's siblings
- never change

That allows react to track siblings that change order or added / removed elements, to optimize it's performance and allow animation framework to track elements.

In this case it's not useful, but react will add a warning in the console and we try to eliminate them.

https://reactjs.org/docs/lists-and-keys.html

Reply:
Okay, changed it to a callback! And I added the keys